internal/proxy/oauthproxy.go

```diff
@@ -193,6 +194,13 @@ func NewOAuthProxy(opts *Options, optFuncs ...func(*OAuthProxy) error) (*OAuthPr
 			return nil, err
 		}
 	}
+	// if allowed EmailDomains or EmailAddresses are defined in the upstream config,
+	// override any previously set emailValidators.
+	if len(p.upstreamConfig.EmailAddresses) != 0 {
+		p.emailValidator = options.NewEmailAddressValidator(p.upstreamConfig.EmailAddresses)
+	} else if len(p.upstreamConfig.EmailDomains) != 0 {
+		p.emailValidator = options.NewEmailDomainValidator(p.upstreamConfig.EmailDomains)
+	}
```

Comment:
Would should supply a defensive closing else statement to make sure we're always passing in a validator.